applications/mne_analyze/plugins/view3d/view3d.cpp

```diff
@@ -218,16 +222,26 @@ QVector<EVENT_TYPE> View3D::getEventSubscriptions(void) const
     temp.push_back(FIDUCIAL_CHANGED);
     temp.push_back(VIEW3D_SETTINGS_CHANGED);
     temp.push_back(SELECTED_MODEL_CHANGED);
+    temp.push_back(MODEL_REMOVED);
     return temp;
 }
 
 //=============================================================================================================
 
-void View3D::updateCoregBem(ANSHAREDLIB::BemDataModel::SPtr pNewModel)
+void View3D::updateCoregBem(QSharedPointer<ANSHAREDLIB::BemDataModel> pNewModel)
 {
-    if(pNewModel->getType() == ANSHAREDLIB_BEMDATA_MODEL) {
+    if(!pNewModel){
+        std::cout << "Empty pointer";
+        return;
+    } else if(!m_p3DModel){
+        std::cout << "Empty model";
```

Comment:
remove std::couts

Reply:
Would it be better to add qWarnings here instead or to remove them completley?

Reply:
yes warnings would be better